### ISSUE TITLE
There are no packages available for installation - Package Control of Sublime Text 3 

### ISSUE THREAD

**Issue by billyhalim** [wbond/package_control#959]
Hello,
I have the following problems :

> Package Control: Attempting to use Urllib downloader due to WinINet error: Error downloading channel. (errno 12045) during HTTP write phase of downloading https://packagecontrol.io/channel_v3.json.
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Resolved IP: 50.116.33.29
  Timeout: 30
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at C:\Documents and Settings\Billy\Application Data\Sublime Text 3\Packages\User\Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
  CA certs file at C:\Documents and Settings\Billy\Application Data\Sublime Text 3\Packages\User\Package Control.merged-ca-bundle
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:550))) downloading https://packagecontrol.io/channel_v3.json.
Package Control: The dependency specified, bz2, is not available
Package Control: Skipping automatic upgrade, last run at 2015-06-17 14:13:08, next run at 2015-06-17 15:13:08 or after
Package Control: Attempting to use Urllib downloader due to WinINet error: Error downloading channel. (errno 12045) during HTTP write phase of downloading https://packagecontrol.io/channel_v3.json.
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Resolved IP: 50.116.33.29
  Timeout: 30
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at C:\Documents and Settings\Billy\Application Data\Sublime Text 3\Packages\User\Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
  CA certs file at C:\Documents and Settings\Billy\Application Data\Sublime Text 3\Packages\User\Package Control.merged-ca-bundle
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:550))) downloading https://packagecontrol.io/channel_v3.json.
error: Package Control
There are no packages available for installation

I also have added "debug" : true to the package setting-user of package control. Are there any solution for that ? I have been stucked. I want to code using Pyhton + Django. If there are any alternative, just mention it. The important thing is I am able to code with Python Django.

Thanks

**Comment by wbond** [comment]
Can you visit https://packagecontrol.io in your browser and post a screenshot of the secure certificate information?

**Comment by billyhalim** [comment]
![untitled](https://cloud.githubusercontent.com/assets/12932465/8250051/5fb5e554-169a-11e5-8013-8e2564d01f1a.PNG)

**Comment by wbond** [comment]
Hmm, something seems weird about your connection to the site. The URL bar should have the padlock in green and say "Codex Non Sufficit LC". The cert fingerprint and serial number look correct, but something else is weird.

What version of Windows XP are you using? Do you have a newer version of Windows on a different machine that you can test? Or possibly an OS X or Linux machine?

**Comment by SalGnt** [comment]
I have the same problem using ST Version 3083 (Stable) on OS X Yosemite (Version 10.10.3).

![error](https://cloud.githubusercontent.com/assets/884093/8267441/441fe72c-1762-11e5-9815-9d9af62960c5.png)

```
Package Control: Error downloading channel. URL error unknown error (_ssl.c:2228) downloading https://packagecontrol.io/channel_v3.json.
error: Package Control
```

When I visit the website the URL bar looks correct.

![pc-io](https://cloud.githubusercontent.com/assets/884093/8267437/17aecc4e-1762-11e5-94f9-d9b7811f05f2.png)

**Comment by billyhalim** [comment]
@wbond Hello, I am using my old computer, with Windows XP SP3 32-bit operating system

**Comment by wbond** [comment]
@SalGnt Please see https://github.com/wbond/package_control/issues/957

**Comment by wbond** [comment]
@billyhalim Can you post a screenshot with the "Page Info" window visible from Firefox? Also, what version of Firefox are you using?

**Comment by SalGnt** [comment]
@wbond Got it, thanks!

**Comment by billyhalim** [comment]
![untitled](https://cloud.githubusercontent.com/assets/12932465/8346786/eabeee3a-1b2b-11e5-8428-f0a62696e7de.PNG)

I use the latest version of Firefox, Firefox 38.0.5

**Comment by wbond** [comment]
@billyhalim How about the security tab in that Page Info dialog? I am trying to figure out why you are getting a grey padlock instead of a green one. It may be related to the error you are seeing in Package Control.

**Comment by billyhalim** [comment]
![untitled](https://cloud.githubusercontent.com/assets/12932465/8350895/88b553e6-1b53-11e5-87d7-c175ae2a3570.PNG)

There is the security tab :)

**Comment by wbond** [comment]
That is strange that it says the website does not provide ownership information. Here is a screenshot of what I see in Firefox:

![packagecontrol io_info](https://cloud.githubusercontent.com/assets/41734/8355520/ce5e752a-1b1d-11e5-905a-42aa8d9bc1bd.png)

**Comment by billyhalim** [comment]
@wbond What should there said ? It seems your screenshot is broken

**Comment by wbond** [comment]
On the security tab the line that says **Owner**, it should say "Codex Non Sufficit LC".

**Comment by wbond** [comment]
@billyhalim Are sites blocked on your connection to the internet?

**Comment by billyhalim** [comment]
hmm...I think no..sites are not blocked @wbond

**Comment by xPea** [comment]
Did you find a solution?

**Comment by wbond** [comment]
No, because the root cause was never found. My hunch is that it is somehow related to Windows XP.

**Comment by xPea** [comment]
I'm using Windows 7 and i got the same problem. Any ideas?

> Package Control: Skipping automatic upgrade, last run at 2015-07-15 07:07:31, next run at 2015-07-15 08:07:31 or after
> Package Control: Attempting to use Urllib downloader due to WinINet error: Error downloading channel. (errno 12045) during HTTP write phase of downloading https://packagecontrol.io/channel_v3.json.
> Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:550))) downloading https://packagecontrol.io/channel_v3.json.
> error: Package Control
> 
> There are no packages available for installation


![1](https://cloud.githubusercontent.com/assets/13328499/8691199/d5e9e228-2abf-11e5-8fbb-ee3d3edee543.PNG)
![2](https://cloud.githubusercontent.com/assets/13328499/8691215/ef3430d0-2abf-11e5-8fa7-95299815b860.PNG)

**Comment by elcaesar** [comment]
Hi, it happened to me the same in Windows. What I did was lower the level of UAC (User Account Control) and run the program in administrator mode No. I hope you serve.

**Comment by holly-juan** [comment]
yesterday i also got the problem "package control There are no packages available for installation",then i found i opened the agent of visiting abroad website,when i closed it ,the problem was solved.

**Comment by expected-value** [comment]
Hi, I've the same error prompt

My settings:
I run windows 7 professional SP 1
Latest chrome version: (44.0.2403.155 m)
Latest sublime version: Build 3083
![ps-ssl-connect-spc fw](https://cloud.githubusercontent.com/assets/13832167/9303984/98c2bb48-44e3-11e5-979f-e9d2fede2bc6.png)

Hope this information helps!

Regards,
T

**Comment by DeveloperZhou** [comment]
Me too. I DONT KNOW WHY

**Comment by wbond** [comment]
I'll need more than just a screenshot to be of any help. Please be sure to create a debug log and post it also. @DeveloperZhou @expected-value

**Comment by DeveloperZhou** [comment]
Oh,sorry.I have tried several versions. Finally, I made it.Thank you anyway. :)

**Comment by Danilux** [comment]
I'm getting the same error, I checked the secure certificate and is secure, I'm using this:
mklink /D "Installed Packages" "C:\path\to\Dropbox\settings\sublime\Installed Packages"
mklink /D "Packages" "C:\path\to\Dropbox\settings\sublime\Packages"
to save the settings to dropbox BTW in case this is relevant.


plugins loaded
schemr ready
Emmet: No need to update PyV8
Could not import subprocess32 module, falling back to subprocess module
Package Control: Installing 1 upgrades
reloading Packages/User/Preferences.sublime-settings
ignoring incompatible BracketHighlighter package, please update to a newer version of BracketHighlighter
ignored packages updated to: ["SideBarEnhancements", "CSS3", "SublimeLinter", "Vintage", "HTML5", "Jekyll", "Markdown", "Theme - Centurion", "Predawn", "Theme - itg.flat", "Theme - Flatland", "BracketHighlighter"]
unloading plugin BracketHighlighter.bh_core
reloading Packages/BracketHighlighter/bh_core.sublime-settings
unloading plugin BracketHighlighter.bh_logging
unloading plugin BracketHighlighter.bh_plugin
unloading plugin BracketHighlighter.bh_regions
unloading plugin BracketHighlighter.bh_remove
unloading plugin BracketHighlighter.bh_rules
unloading plugin BracketHighlighter.bh_search
unloading plugin BracketHighlighter.bh_swapping
unloading plugin BracketHighlighter.bh_wrapping
unloading plugin BracketHighlighter.ure
top level value must be an array
error: Package Control

There are no packages available for installation
Exception in thread Thread-6:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "./threading.py", line 858, in run
  File "package_control.automatic_upgrader in C:\Users\Danilux\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 198, in do_upgrades
  File "package_control.package_manager in C:\Users\Danilux\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 749, in install_package
PermissionError: [WinError 32] El proceso no tiene acceso al archivo porque está siendo utilizado por otro proceso: 'C:\\Users\\Danilux\\AppData\\Roaming\\Sublime Text 3\\Installed Packages\\Package Control.sublime-package'

reloading Packages/User/Package Control.sublime-settings
error: Package Control

There are no packages available for installation
error: Package Control

There are no packages available for installation
reloading Packages/User/Preferences.sublime-settings
error: Package Control

There are no packages available for installation

**Comment by nilsborchers** [comment]
Guys I have the same issue.. @wbond about the blocked sites thing: I'm in Singapore, they block some sites but not packagecontrol.io
I did try deleting the ca file, and it got regenerated (or re-appeared somehow :)) but that didnt change anything ... still getting: "there are no packages available for installation"

Am on Yosemite, Sublime Build 3083

**Comment by wbond** [comment]
So the thing is "no packages available for installation" just means *something* went wrong. In most situations, it is different things going wrong. This is why posting a debug log is important. It shows what is going on behind the scenes.

**Comment by nilsborchers** [comment]
oops I sorry .)
here goes:

```
reloading Packages/User/Package Control.sublime-settings
Package Control: Settings change detected, clearing cache
Package Control: Fetching list of available packages
  Platform: osx-x64
  Sublime Text Version: 3083
  Package Control Version: 2.0.0
Package Control: Download Debug
  URL: https://sublime.wbond.net/channel.json
  Resolved IP: 50.116.34.243
  Timeout: 30
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/nilsborchers/Library/Application Support/Sublime Text 3/Packages/User/Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to sublime.wbond.net on port 443
Package Control: Urllib HTTPS Debug General
  Connecting to sublime.wbond.net on port 443
  CA certs file at /Users/nilsborchers/Library/Application Support/Sublime Text 3/Packages/User/Package Control.system-ca-bundle
  Successfully upgraded connection to sublime.wbond.net:443 with SSL
  Server SSL certificate:
    subject: CN=sublime.wbond.net,OU=Domain Control Validated - RapidSSL(R),OU=See www.rapidssl.com/resources/cps (c)15,OU=GT52951423,serialNumber=f42O34HKwvQTRlRAsXiRPvAWUUwbo22p
    common name: sublime.wbond.net
    expire date: Mar 29 02:42:51 2016 GMT
  Certificate validated for sublime.wbond.net
Package Control: Urllib HTTPS Debug Write
  GET /channel.json HTTP/1.1
  User-Agent: Sublime Package Control v2.0.0
  Host: sublime.wbond.net
  If-Modified-Since: Wed, 02 Sep 2015 16:05:02 GMT
  Accept-Encoding: gzip,deflate
  Connection: Keep-Alive
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: nginx
  Date: Wed, 02 Sep 2015 16:29:06 GMT
  Last-Modified: Wed, 02 Sep 2015 16:05:02 GMT
  Connection: keep-alive
  ETag: "55e71e2e-36c"
Package Control: Using cached content for https://sublime.wbond.net/channel.json
Package Control: Found git at "/usr/bin/git"
Package Control: Found git at "/usr/bin/git"
Package Control: Trying to execute command /usr/bin/git symbolic-ref -q HEAD
Package Control: Trying to execute command /usr/bin/git config --get branch.python3.remote
Package Control: Trying to execute command /usr/bin/git config --get branch.python3.merge
Package Control: Trying to execute command /usr/bin/git fetch origin
Package Control: Trying to execute command /usr/bin/git log ..origin/python3
error: Package Control

There are no packages available for installation
Package Control: Urllib HTTPS Debug General
  Closing connection to sublime.wbond.net on port 443 after 1 request
```

**Comment by wbond** [comment]
@nilsborchers You are running a really old version of Package Control which is no longer supported, hence there are no packages that can be installed. Follow the directions at https://packagecontrol.io/docs/troubleshooting#Purging_and_Reinstalling to upgrade.

**Comment by nilsborchers** [comment]
@wbond thanks so much man ... not it says 3.0.1 which is probably a lot better right? Anyway it works now.
Does package control usually update itself?

**Comment by FichteFoll** [comment]
Yes

**Comment by wbond** [comment]
Yes, there was just a bug in some 2.0.0 alphas (for ST3 support) that prevented automatic upgrades.

**Comment by wbond** [comment]
@billyhalim Sorry for taking to long to try and come up with a solution. I think what is going on here is that Sublime Text 2 on Windows compiled with an old version of OpenSSL for the `_ssl` Python module. The side effect of this is that the modern, SHA-2 certificate that `packagecontrol.io` uses is not recognized.

Could you try opening `C:\Documents and Settings\Billy\Application Data\Sublime Text 3\Packages\` in Windows Explorer and deleting the `Package Control` and any `0_package_control_loader` folders? Then follow the directions at https://packagecontrol.io/installation#st2 and see if it works.

If it does not, can you provide a debug log from the session when you run the install code?

**Comment by Reedyseth** [comment]
@wbond 


> Could you try opening C:\Documents and Settings\Billy\Application Data\Sublime Text 3\Packages\ in Windows Explorer and deleting the Package Control and any 0_package_control_loader folders? Then follow the directions at https://packagecontrol.io/installation#st2 and see if it works.

i just followed this steps and it didn't work.
I am working under windows 7 64bit with Sublime Text 2, this is my log:

```
Package Control: The dependency specified, select-windows, is not available
Package Control: The dependency specified, bz2, is not available
Package Control: No updated packages
error: Package Control

There are no packages available for installation
```

**Comment by Gutem** [comment]
I'm getting this:
```
startup, version: 3083 osx x64 channel: stable
...
Package Control: Skipping automatic upgrade, last run at 2015-09-13 13:17:54, next run at 2015-09-13 14:17:54 or after
Package Control: Error downloading channel. URL error _ssl.c:534: The handshake operation timed out downloading https://packagecontrol.io/channel_v3.json.
error: Package Control

There are no packages available for installation
```
Trying to connect  to `https://packagecontrol.io/` I'm getting:
![screenshot 2015-09-13 14 18 16](https://cloud.githubusercontent.com/assets/342215/9837848/729c95ca-5a22-11e5-8184-2e0cee500ef0.png)

**Comment by wbond** [comment]
@Reedyseth You'll need to turn on the `debug` setting: https://packagecontrol.io/docs/troubleshooting#Enabling_the_Debug_Log

**Comment by wbond** [comment]
@Gutem Something on your network seems to be messing with your secure connection. I can tell you packagecontrol.io only accepts TLS 1.x, not SSL v3  (or v2). Perhaps there is something on your network that can't handle TLS 1 handshakes?

**Comment by Gutem** [comment]
@wbond I think it's something with my ISP. Back to normal now, without any modification... 
Tks

**Comment by Reedyseth** [comment]
> @Reedyseth You'll need to turn on the debug setting: https://packagecontrol.io/docs/troubleshooting#Enabling_the_Debug_Log

@wbond I have enabled the debug mode on the Package Control Settings not the Package Control User settings. Nothings was outputted. For now I will double check the secure connection and notify.
Regards.!!

**Comment by astrism** [comment]
I'm having this issue as well, my debug looks like this:
```
Package Control: Fetching list of available packages
  Platform: osx-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/xx/Library/Application Support/Sublime Text 3/Packages/User/Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/xx/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
  Successfully upgraded connection to packagecontrol.io:443 with SSL Using TLSv1/SSLv3 with cipher DHE-RSA-AES128-SHA
  Server SSL certificate:
    subject: CN=codexns.io,O=Codex Non Sufficit LC,L=Newbury,S=Massachusetts,C=US,serialNumber=471714639,businessCategory=Private Organization,incorporationState=Massachusetts,incorporationCountry=US
    subject alt name: dev.codexns.io, rc.codexns.io, packagecontrol.io, wbond.net, codexns.io
    expire date: Oct 29 23:59:59 2015 GMT
  Certificate validated for packagecontrol.io
Package Control: Urllib HTTPS Debug Write
  GET /channel.json HTTP/1.1
  User-Agent: Package Control v3.1.2
  Accept-Encoding: bzip2,gzip,deflate
  Host: packagecontrol.io
  If-Modified-Since: Wed, 30 Sep 2015 21:05:04 GMT
  Connection: Keep-Alive
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: nginx
  Date: Wed, 30 Sep 2015 21:29:13 GMT
  Last-Modified: Wed, 30 Sep 2015 21:05:04 GMT
  Connection: keep-alive
  ETag: "560c4e80-371"
  Strict-Transport-Security: max-age=15552000
Package Control: Using cached content for https://packagecontrol.io/channel.json from /Users/xx/Library/Application Support/Sublime Text 3/Packages/User/Package Control.cache/a67878a45ebe1d30b0f6a9dc5fbedd5a
Package Control: Download Debug
  URL: https://packagecontrol.io/channel.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib HTTPS Debug General
  Re-using connection to packagecontrol.io on port 443 for request #2
Package Control: Urllib HTTPS Debug Write
  GET /channel.json HTTP/1.1
  User-Agent: Package Control v3.1.2
  Accept-Encoding: bzip2,gzip,deflate
  Host: packagecontrol.io
  If-Modified-Since: Wed, 30 Sep 2015 21:05:04 GMT
  Connection: Keep-Alive
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: nginx
  Date: Wed, 30 Sep 2015 21:29:13 GMT
  Last-Modified: Wed, 30 Sep 2015 21:05:04 GMT
  Connection: keep-alive
  ETag: "560c4e80-371"
  Strict-Transport-Security: max-age=15552000
Package Control: Using cached content for https://packagecontrol.io/channel.json from /Users/xx/Library/Application Support/Sublime Text 3/Packages/User/Package Control.cache/a67878a45ebe1d30b0f6a9dc5fbedd5a
```

**Comment by wbond** [comment]
@astrism Check your Preferences > Package Settings > Package Control > Settings - User and delete the key `channels`. `https://packagecontrol.io/channel.json` is for Package Control 2.0, but you are using Package Control 3.1. Once you delete the `channels` key, it will revert to using the default of `https://packagecontrol.io/channel_v3.json`.

**Comment by astrism** [comment]
Thanks @wbond, deleting the channels key worked!

**Comment by gudstrand** [comment]
I have followed all the advice in this thread but I am still getting an error.

Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:550))) downloading https://packagecontrol.io/channel_v3.json.

I can go to the site without issue using chrome and I can even download the channel_v3.json file.  Any ideas?

**Comment by gudstrand** [comment]
Here is the debug log:

WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED False
Package Control: Fetching list of available packages
  Platform: osx-x64
  Sublime Text Version: 3095
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib Debug Proxy
  http_proxy: digitalwind.net:22
  https_proxy: <host and port>
  proxy_username: <real user>
  proxy_password: <correct password>
Package Control: Found previously exported CA bundle at /Users/gary/Library/Application Support/Sublime Text 3/Packages/User/Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to <host and port>
Package Control: Urllib HTTPS Debug Write
  CONNECT packagecontrol.io:443 HTTP/1.1
  Host: packagecontrol.io:443
  User-Agent: Package Control v3.1.2
  Proxy-Connection: Keep-Alive
Package Control: Error downloading channel. HTTP exception BadStatusLine (SSH-2.0-OpenSSH_5.3
) downloading https://packagecontrol.io/channel_v3.json.

**Comment by gudstrand** [comment]
I am using a proxy, I put placeholders in the fields but it looks like they were blanked.  I did verify that I can establish a proxy connection using those settings.  But I still get the error downloading the package control file.

**Comment by gudstrand** [comment]
Update, I took my laptop to another network that did not require a proxy and it worked.  Seems there is something wrong with the proxy code in the package manager.  I am glad that I was able to install the package I needed but it is unfortunate that I cannot use it when behind a proxy (which I am).

**Comment by Reedyseth** [comment]
> I am glad that I was able to install the package I needed but it is unfortunate that I cannot use it when behind a proxy (which I am).

@gudstrand Perhaps you did not edit the correct file. Go go to `Preferences > Package Settings > Package Control > Settings user` and add the proxy values like this:
```javascript
{
	"proxy_username": "",
	"proxy_password": "",
	"http_proxy": "proxy-url",
	"https_proxy": "https-proxy-url",
}
```
Fill the user and password if necessary other wise delete it. I just hope that you are not under an ISA server and with the https protocol firewalled.

Regards.!!

**Comment by gudstrand** [comment]
On Mon, Oct 12, 2015 at 4:31 PM, Israel Barragan C. <
notifications@github.com> wrote:

> "proxy_username": "", "proxy_password": "", "http_proxy": "proxy-url", "
> https_proxy": "https-proxy-url",


​
Thanks for the help, I had that same entry before and just tried it again
and it failed.  I have two proxys configured (work just fine from my mac)
and neither work with package control.   Wished it did, it would make
getting new packages much simpler.​


—
Gary

​

**Comment by Reedyseth** [comment]
> I have two proxys configured (work just fine from my mac)
> and neither work with package control.   Wished it did, it would make
> getting new packages much simpler.​

If you are under an ISA server you can instal [CNTLM](http://cntlm.sourceforge.net/) to by pass the entry of username and password, on my case beside the proxy the company also installed a firewall and that was a dead end for me, no more package control and adding packages manually.

I don't want to turn this issue into antoher one. On my case I think it was more issues with the company firewall and for that reason I was not able to list package, although I had the same issue on a computer without a proxy. If I see the same issue I will post it here.

**Comment by gudstrand** [comment]
It is not a ISA server, one proxy is at Hostmonster and the other Digital
Ocean.  I am using Secure Pipes on my Mac to connect to the SOCKS proxies
at those hosts and it works great.   But, adding the same ip addy, user and
password to the package manager config results in an empty install list in
the package manager.

—
Gary Udstrand
*612-670-2163*

*Digitalwind Consulting Inc.  *

[image: Business card]

​

**Comment by wbond** [comment]
@gudstrand Your issue seems distinct from the OP here. Please open a new issue and include a debug log, as instructed by the contribution guidelines link at the top of the new issue form.

Also note, Package Control only supports HTTPS proxies, not SOCKS.

**Comment by koistya** [comment]
Solved this issue buy running Sublime Text 3 as Administrator (on Windows 8.1 x64).

**Comment by fsshenqi** [comment]
Solved this issue(Windows XP and Sublime Text 3 (build 3083))
Open Internet Options in Control Panel ->Advanced tab ->enable the "use TLS 1.0"

**Comment by roshan007** [comment]
@koistya  thanks it resolved the problem.i have mess up with a lot of solution but didn't worked.

**Comment by ccnixon** [comment]
I'm getting this issue as well. I tried a bunch of the fixes but still not working:

```json
reloading Packages/User/Package Control.sublime-settings
Package Control: Settings change detected, clearing cache
reloading Packages/User/Package Control.sublime-settings
reloading Packages/User/Package Control.sublime-settings
Package Control: Fetching list of available packages
  Platform: osx-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/christophernixon/Library/Application Support/Sublime Text 3/Packages/User/Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/christophernixon/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
  Successfully upgraded connection to packagecontrol.io:443 with SSL Using TLSv1/SSLv3 with cipher DHE-RSA-AES128-SHA
  Server SSL certificate:
    subject: CN=codexns.io,O=Codex Non Sufficit LC,L=Newbury,S=Massachusetts,C=US,serialNumber=471714639,businessCategory=Private Organization,incorporationState=Massachusetts,incorporationCountry=US
    subject alt name: codexns.com, packagecontrol.io, wbond.net, rc.codexns.io, codexns.io, www.codexns.io
    expire date: Nov 27 23:59:59 2016 GMT
  Certificate validated for packagecontrol.io
Package Control: Urllib HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  Connection: Keep-Alive
  Accept-Encoding: gzip,deflate
  Host: packagecontrol.io
  User-Agent: Package Control v3.1.2
  If-Modified-Since: Sat, 02 Jan 2016 23:10:01 GMT
Package Control: Urllib HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Sat, 02 Jan 2016 23:24:09 GMT
  Content-Type: application/json
  Last-Modified: Sat, 02 Jan 2016 23:20:01 GMT
  Transfer-Encoding: chunked
  Connection: keep-alive
  Vary: Accept-Encoding
  Strict-Transport-Security: max-age=15552000
  Content-Encoding: gzip
Package Control: Caching https://packagecontrol.io/channel_v3.json in /Users/christophernixon/Library/Application Support/Sublime Text 3/Packages/User/Package Control.cache/01524fae79697630d0454ba3fabd9414
Package Control: Urllib HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
```

I'm using ST3 Build 3083 on Mac OSX 10.11.2

Thanks!

**Comment by weisk** [comment]
Happening to me aswell, as of today, same specs than this guy ^

**Comment by cillosis** [comment]
Just started happening on my end too (ST3 3083 on OSX 10.8.5).

**Comment by weisk** [comment]
yup, using an archived version http://web.archive.org/web/timemap/link/https://packagecontrol.io/channel_v3.json works, but obviously they are probably outdated

**Comment by Trippnology** [comment]
Me too. Looks like it's connecting OK, but the response is lacking any packages:

`{
    "schema_version": "3.0.0",
    "packages_cache": { },
    "dependencies_cache": { },
    "repositories": [ ]
}`

**Comment by jmk2142** [comment]
I'm having the same issue. Looks like channel_v3.json is producing nothing. Used the older channel_v3.json that @weisk posted and package control seems to work. I'm assuming all we can do is wait at this point. I wish I found this thread earlier.

**Comment by schandrac** [comment]
Same issue for me after updating ST3 3082 on OSX 10.11.1

**Comment by ex1tium** [comment]
Also having problems archive link @weisk posted doesn't work here.

**Comment by Trippnology** [comment]
This link is actually an xml file, not json: http://web.archive.org/web/timemap/link/https://packagecontrol.io/channel_v3.json

Save it, and copy the link to the latest archive copy (which is currently http://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json )

Copying the following lines to `Preferences > Package Settings > Package Control > Settings - User` will allow you to install packages again.

```
	"channels":
	[
		"https://packagecontrol.io/channel_v3.json",
		"https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json"
	],
````

**Comment by schandrac** [comment]
Following suggestion by Trippnology worked. Thanks!

> Copying the following lines to Preferences > Package Settings > Package Control > Settings - User will allow you to install packages again.

>    "channels":
    [
        "https://packagecontrol.io/channel_v3.json",
        "https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json"
    ],

**Comment by jelpern** [comment]
This worked for me ^

@Trippnology are there any security implications of getting directed to packages this way?

**Comment by brotherhutch** [comment]
@Trippnology worked - thank you. i take it we're just waiting until the proper channel is populated again.

**Comment by Trippnology** [comment]
Glad this is working for you guys. It's very much just a workaround until the current feed comes back. Shoutout to @weisk for the archive link.

@jelpern It's simply an old copy of the real feed. It is served over https, but there may be other concerns.

**Comment by mlslucio** [comment]
work for me too, thanksTrippnology!

**Comment by oxapentane** [comment]
Same here. JSON received from the site is empty for both v2 and v3 channels.

```JSON
$ curl https://packagecontrol.io/channel_v3.json
{"schema_version": "3.0.0", "packages_cache": {}, "repositories": [], "dependencies_cache": {}}%
$ curl https://packagecontrol.io/channel.json
{"packages_cache": {}, "certs": {}, "repositories": [], "schema_version": "2.0"}%
```

**Comment by yurri87** [comment]
Please help!:

DPI scale: 1
startup, version: 3083 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/Yuriy/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/Yuriy/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/Yuriy/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.142021
startup time: 0.218021
first paint time: 0.225021
unsupported pixel depth
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.quick_panel
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
plugins loaded
Package Control: Download Debug
  URL: https://packagecontrol.io/submit?sublime_platform=windows&package_control_version=3.1.2&sublime_version=3083&package=Package+Control&version=3.1.2&operation=install
  Timeout: 30
  Resolved IP: 50.116.33.29
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Thu, 29 Oct 2015 06:00:00 GMT
    expire date: Mon, 28 Nov 2016 05:59:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /submit?sublime_platform=windows&package_control_version=3.1.2&sublime_version=3083&package=Package+Control&version=3.1.2&operation=install HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  Accept-Encoding: gzip,deflate
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Sun, 03 Jan 2016 14:07:54 GMT
  Content-Type: application/json
  Transfer-Encoding: chunked
  Connection: keep-alive
  Vary: Accept-Encoding
  X-App-Version: 1.3.3
  Strict-Transport-Security: max-age=15552000
  Content-Encoding: gzip
Package Control: Installing 1 missing dependency
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
Package Control: WinINet HTTPS Debug General
  Re-using connection to packagecontrol.io on port 443 for request #1
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  Accept-Encoding: gzip,deflate
  If-Modified-Since: Sun, 03 Jan 2016 14:00:02 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: nginx
  Date: Sun, 03 Jan 2016 14:07:54 GMT
  Last-Modified: Sun, 03 Jan 2016 14:00:02 GMT
  Connection: keep-alive
  ETag: "56892962-5f"
  Strict-Transport-Security: max-age=15552000
Package Control: Using cached content for https://packagecontrol.io/channel_v3.json from C:\Users\Yuriy\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.cache\01524fae79697630d0454ba3fabd9414
Package Control: Download Debug
  URL: https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 92.255.241.100
Package Control: Attempting to use Urllib downloader due to WinINet error: Error downloading channel. Secure channel error, server not providing SSL (errno 12157) during HTTP write phase of downloading https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json.
Package Control: Download Debug
  URL: https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 92.255.241.100
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at C:\Users\Yuriy\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to web.archive.org on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at C:\Users\Yuriy\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.merged-ca-bundle
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host web.archive.org returned an invalid certificate ([SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:550))) downloading https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json.
Package Control: The dependency specified, bz2, is not available
Package Control: Skipping automatic upgrade, last run at 2016-01-03 19:24:50, next run at 2016-01-03 20:24:50 or after
Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 2 requests
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 92.255.241.100
Package Control: Attempting to use Urllib downloader due to WinINet error: Error downloading channel. Secure channel error, server not providing SSL (errno 12157) during HTTP write phase of downloading https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json.
Package Control: Download Debug
  URL: https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 92.255.241.100
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at C:\Users\Yuriy\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.system-ca-bundle
Package Control: Urllib HTTPS Debug General
  Connecting to web.archive.org on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at C:\Users\Yuriy\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.merged-ca-bundle
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host web.archive.org returned an invalid certificate ([SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:550))) downloading https://web.archive.org/web/20150905194312/https://packagecontrol.io/channel_v3.json.

Windows 7 x64

**Comment by nurdeen** [comment]
Yes I receive empty JSON like xapentane

**Comment by br3ad** [comment]
Hello,@wbond

**I've had some problems：**

![222](https://cloud.githubusercontent.com/assets/8861289/12079159/541830dc-b26a-11e5-9d18-6a12196897b6.png)

**This is debug log：**
```
DPI scale: 1
startup, version: 3083 windows x64 channel: stable
executable: /D/Development tools/sublime text 3/sublime_text.exe
working dir: /D/Development tools/sublime text 3
packages path: /C/Users/1/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/1/AppData/Roaming/Sublime Text 3/Local
zip path: /D/Development tools/sublime text 3/Packages
zip path: /C/Users/1/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.244648
startup time: 0.347649
first paint time: 0.348648
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.quick_panel
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.02-bz2
reloading plugin AdvancedNewFile.AdvancedNewFile
reloading plugin AutoFileName.autofilename
reloading plugin AutoFileName.getimageinfo
reloading plugin Colorsublime.colorsublime-plugin
reloading plugin DocBlockr.jsdocs
reloading plugin Emmet.emmet-plugin
reloading plugin Javap.Jar
reloading plugin Javap.Javap
reloading plugin Nodejs.__init__
reloading plugin Nodejs.Nodejs
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin PackageResourceViewer.package_resource_viewer
reloading plugin PackageResourceViewer.package_resources
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SublimeGit.SublimeGit
reloading plugin sublimelint.commands
reloading plugin sublimelint.sublimelint
reloading plugin SublimeLinter-csslint.linter
SublimeLinter: csslint linter loaded 
reloading plugin SublimeLinter-html-tidy.linter
SublimeLinter: htmltidy linter loaded 
reloading plugin SublimeLinter-javac.linter
SublimeLinter: javac linter loaded 
reloading plugin View In Browser.ViewInBrowserCommand
reloading plugin Autoprefixer.Autoprefixer
reloading plugin Autoprefixer.node_bridge
reloading plugin ColorPicker.sublimecp
reloading plugin IMESupport.imesupportplugin
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
plugins loaded
SublimeLinter: debug mode: off 
Emmet: No need to update PyV8
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Exception in thread Thread-6:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Exception in thread Thread-9:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Thu, 29 Oct 2015 08:00:00 GMT
    expire date: Mon, 28 Nov 2016 07:59:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: "56892962-6e"
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Sun, 03 Jan 2016 14:00:02 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: nginx
  Date: Sun, 03 Jan 2016 14:07:36 GMT
  Content-Type: application/json
  Last-Modified: Sun, 03 Jan 2016 14:00:02 GMT
  Connection: keep-alive
  ETag: "56892962-6e"
  Content-Encoding: bzip2
  Vary: Accept-Encoding
  Strict-Transport-Security: max-age=15552000
Package Control: Using cached content for https://packagecontrol.io/channel_v3.json from C:\Users\1\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.cache\01524fae79697630d0454ba3fabd9414
Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Package Control: No updated packages
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Thu, 29 Oct 2015 08:00:00 GMT
    expire date: Mon, 28 Nov 2016 07:59:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: "56892962-6e"
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Sun, 03 Jan 2016 14:00:02 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: nginx
  Date: Sun, 03 Jan 2016 14:07:46 GMT
  Content-Type: application/json
  Last-Modified: Sun, 03 Jan 2016 14:00:02 GMT
  Connection: keep-alive
  ETag: "56892962-6e"
  Content-Encoding: bzip2
  Vary: Accept-Encoding
  Strict-Transport-Security: max-age=15552000
Package Control: Using cached content for https://packagecontrol.io/channel_v3.json from C:\Users\1\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.cache\01524fae79697630d0454ba3fabd9414
Exception in thread Thread-11:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
Exception in thread Thread-12:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Exception in thread Thread-14:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Exception in thread Thread-15:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Exception in thread Thread-16:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Exception in thread Thread-17:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Package Control: Fetching list of available packages
  Platform: windows-x64
  Sublime Text Version: 3083
  Package Control Version: 3.1.2
Exception in thread Thread-19:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: [Errno 11004] getaddrinfo failed
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, Massachusetts, Private Organization, 471714639, US, Massachusetts, Newbury, Codex Non Sufficit LC, codexns.io
    issuer: US, GeoTrust Inc., GeoTrust EV SSL CA - G4
    common name: codexns.io
    issue date: Thu, 29 Oct 2015 08:00:00 GMT
    expire date: Mon, 28 Nov 2016 07:59:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  User-Agent: Package Control v3.1.2
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: "56892962-6e"
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Sun, 03 Jan 2016 14:00:02 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Sun, 03 Jan 2016 14:33:08 GMT
  Content-Type: application/json
  Content-Length: 106
  Last-Modified: Sun, 03 Jan 2016 14:30:01 GMT
  Connection: keep-alive
  ETag: "56893069-6a"
  Content-Encoding: bzip2
  Vary: Accept-Encoding
  Strict-Transport-Security: max-age=15552000
  Accept-Ranges: bytes
Package Control: Caching https://packagecontrol.io/channel_v3.json in C:\Users\1\AppData\Roaming\Sublime Text 3\Packages\User\Package Control.cache\01524fae79697630d0454ba3fabd9414
Exception in thread Thread-21:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "SublimeGit/sgit/status.py", line 561, in run
  File "SublimeGit/sgit/cmd.py", line 206, in git_string
  File "SublimeGit/sgit/cmd.py", line 34, in _string
ValueError: need more than 2 values to unpack

Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
```

**My settings:**

I run windows：Windows 7 x64
Browser Version：Google Chrome  47
Latest sublime version: Build 3083Latest chrome version
![33](https://cloud.githubusercontent.com/assets/8861289/12079153/beddb564-b269-11e5-9290-2c0b618e6890.png)

Is there any solution ? Thx :)

**Comment by yingogobot** [comment]
same issue here, please advise.

**Comment by phalvinayak** [comment]
Same Issue Here... Please help... :(

**Comment by isaactzab** [comment]
I have the same issue here. I've deleted and reinstalled manually the package manager and still not working. Is the repository working?

**Comment by isaactzab** [comment]
Looks like package control repository server is down for now. 
Seemingly server is hosted by linode in atlanta, and the servers are experiencing DDoS attack. 
Check the status of the server.
https://linode.statuspage.io/

(https://stackoverflow.com/questions/34563717/packagecontrol-io-is-not-working-how-can-i-install-
package-control-and-install)

**Comment by orbitbot** [comment]
Guys, anyone having problems because of the DDoS attack should look at the workaround posted by @Trippnology above, https://github.com/wbond/package_control/issues/959#issuecomment-168447848 , works fine for me.

**Comment by patik** [comment]
> Guys, anyone having problems because of the DDoS attack should look at the workaround posted by @Trippnology above, [#959 (comment)](https://github.com/wbond/package_control/issues/959#issuecomment-168447848) , works fine for me.

That workaround will only list and install older versions of the packages from September 2015 or earlier (when the archive.org copy of the JSON file was made, I presume). 

Even if you try to upgrade a package you will not get a newer one unless you bypass Package Control and install the package manually.

**Comment by wbond** [comment]
For issues related to the recent outage, please see #1057

**Comment by heihachi88** [comment]
@Trippnology it does work, thanks. Any drawbacks when hardcoding the path in a config file?

**Comment by Trippnology** [comment]
@heihachi88 Yes, you will have to remember to remove it again, once the official channel comes back online.

**Comment by stanwu** [comment]
It seems happen again XD
https://packagecontrol.io/channel_v3.json  returns
{"packages_cache": {}, "repositories": [], "schema_version": "3.0.0", "dependencies_cache": {}}

**Comment by yoki123** [comment]
and https://packagecontrol.io/channel.json returns
{"schema_version": "2.0", "certs": {}, "packages_cache": {}, "repositories": []}

**Comment by wbond** [comment]
I'm locking this and asking everyone having issues related to the channel and availability to see #1057

**Comment by wbond** [comment]
I'm closing this issue as it is a grab bag of a bunch of different issues, many of which have been solved, and others which have solutions users need to do some work to accomplish. In summary:

 - There was an outage in early January that resulted in the error message for many users
 - The OP appeared to have a proxy on his network intercepting SSL traffic (windows error 12045 and OpenSSL error) - there is a new issue to track an enhancement for debugging this: https://github.com/wbond/package_control/issues/1131
 - Some users were using old channel URLs, this was fixed by (8b0c1fe3b64fb3ccb8f30d2a5652a3f8d8f4600f)
 - Some users had TLSv1 disabled in Internet Explorer preferences (on Windows XP)
 - SOCKS proxies are not supported
